cloudinit/sources/DataSourceAzure.py

```diff
@@ -1468,23 +1484,12 @@ def _generate_network_config_from_imds_metadata(imds_metadata) -> dict:
 
 @azure_ds_telemetry_reporter
 def _generate_network_config_from_fallback_config() -> dict:
```

Comment:
is there a reason why this function isn't in the DataSource class? it'll make refactoring it to take a networking object much harder

Reply:
I'm not sure. I think it was part of a refactoring recently from parse_network_config and parse_network_config has always been outside of the datasource class.